cocos/base/CCRef.cpp

```diff
@@ -111,7 +116,7 @@ void Ref::release()
     --_referenceCount;
 
 #if CC_ENABLE_SCRIPT_BINDING && CC_ENABLE_GC_FOR_NATIVE_OBJECTS
-    if (_scriptOwned && _referenceCount==1 && _rooted)
+    if (_scriptOwned && _rooted && _referenceCount==_referenceCountAtRootTime)
```

Comment:
But we don't know the retain release order, it may be retained by A then B then C, and released in order of B A C, I can't see why it's useful.